doc/man3/SSL_CTX_set_tmp_ecdh.pod

```diff
@@ -0,0 +1,48 @@
+=pod
+
+=head1 NAME
+
+SSL_CTX_set_tmp_ecdh, SSL_set_tmp_ecdh, SSL_CTX_set_ecdh_auto, SSL_set_ecdh_auto
+- handle ECDH keys for ephemeral key exchange
+
+=head1 SYNOPSIS
+
+#include <openssl/ssl.h>
+
+long SSL_CTX_set_tmp_ecdh(SSL_CTX *ctx, const EC_KEY *ecdh);
+long SSL_set_tmp_ecdh(SSL *ssl, const EC_KEY *ecdh);
+
+long SSL_CTX_set_ecdh_auto(SSL_CTX *ctx, int state);
+long SSL_set_ecdh_auto(SSL *ssl, int state);
+
+=head1 DESCRIPTION
+
+SSL_CTX_set_tmp_ecdh() sets ECDH parameters to be used to be B<ecdh>.
+The key is inherited by all B<ssl> objects created from B<ctx>.
+
+SSL_set_tmp_ecdh() sets the parameters only for B<ssl>.
```

Comment:
Both of these functions are provided for backwards compatibility only and should not be used in new code. We should probably formally deprecate them.

Reply:
Since these are macros and not functions I am not sure how to mark them as deprecated. Also do we have a replacement or is it a functionality that is not needed anymore?

Reply:
The replacement is `SSL_CTX_set1_groups()` and `SSL_set1_groups`.

To deprecate macros simply enclose the definitions in a `#if OPENSSL_API_COMPAT < 0x10200000L` block.